eip_operator/src/controller/node.rs

```diff
@@ -71,10 +86,26 @@ impl k8s_controller::Context for Context {
         if eip_description.network_interface_id != Some(eni_id.to_owned())
             || eip_description.private_ip_address != Some(node_ip.to_owned())
         {
-            crate::aws::associate_eip(&self.ec2_client, allocation_id, &eni_id, node_ip).await?;
+            match crate::eip::set_status_attached(&eip_api, &eip, &eni_id, node_ip, name).await {
+                Ok(_) => {
+                    info!("Found matching Eip, claiming it");
+                    crate::aws::associate_eip(&self.ec2_client, allocation_id, &eni_id, node_ip)
+                        .await?;
+                }
+                Err(err)
+                    if err
+                        .to_string()
+                        .contains("Operation cannot be fulfilled on eips.materialize.cloud") =>
+                {
+                    info!(
+                        "Pod {} failed to claim eip {}, rescheduling to try another",
+                        name, eip_name
+                    );
+                    return Ok(Some(Action::requeue(Duration::from_secs(1))));
```

Comment:
why do we have this special handling at all? if the reconcile loop returns an error, it will already log an error and schedule a rereconcile in the very near future - i don't quite understand what we gain from handling this manually

Reply:
If we're expecting this as a valid possibility due to a conflict and we're able to appropriately handle it is there really an error? I can just return the error, but I think it makes logs more confusing, than having an info message that indicates this sort of thing was expected.